gps/gpsparser.go

```diff
@@ -100,7 +112,10 @@ func findTime(val string) time.Time {
 	h, _ := strconv.ParseInt(val[0:2], 10, 8)
 	m, _ := strconv.ParseInt(val[2:4], 10, 8)
 	s, _ := strconv.ParseInt(val[4:6], 10, 8)
-	ms, _ := strconv.ParseInt(val[7:10], 10, 16)
+	ms := int64(0)
+	if len(val) == 10 {
```

Comment:
```suggestion
	if len(val) >= 10 {
```
It's OK for `val` to be longer or we are working with just the block between two first commas here?

Reply:
Improved that slice expression, good catch!